src/sparseml/pytorch/recipe_template/main.py

```
@@ -0,0 +1,351 @@
+        final_lr="eval(final_lr)",
+    )
+    return [epoch_modifier, lr_modifier]
+
+
+def _get_pruning_modifier_builders(
+    pruning_algo: str,
+    model: Optional[Module] = None,
+    global_sparsity: bool = False,
+) -> Tuple[List[ModifierYAMLBuilder], Dict[str, Any]]:
+    pruning_args = dict(
+        start_epoch=0.0,
+        params="__ALL_PRUNABLE__"
+        if model is None
+        else _get_prunable_params_for_module(model),
+        init_sparsity=0.05,
+        final_sparsity=0.8,
+        end_epoch=10.0,
+        update_frequency=1.0,
+        mask_type="eval(mask_type)",
+        global_sparsity="eval(global_sparsity)",
+        leave_enabled=True,
+        inter_func="cubic",
+    )
+    extra_recipe_variables = {}
+
+    if pruning_algo == "magnitude":
+        modifier_class = (
+            GlobalMagnitudePruningModifier
+            if global_sparsity
+            else MagnitudePruningModifier
+        )
+        # magnitude pruning and global magnitude pruning have
+        # their own defaults for global sparsity
+        del pruning_args["global_sparsity"]
+
+    elif pruning_algo == "acdc":
+        modifier_class = ACDCPruningModifier
+
+        # AC/DC algorithm only specifies a final compression sparsity
+        pruning_args["compression_sparsity"] = pruning_args["final_sparsity"]
+
+        del pruning_args["init_sparsity"]
+        del pruning_args["final_sparsity"]
+        del pruning_args["inter_func"]
+
+    elif pruning_algo == "mfac":
+        modifier_class = MFACPruningModifier
+        extra_recipe_variables.update(dict(num_grads=64, fisher_block_size=10000))
+        pruning_args.update(
+            dict(
+                num_grads="eval(num_grads)", fisher_block_size="eval(fisher_block_size)"
+            )
+        )
+
+    elif pruning_algo == "movement":
+        modifier_class = MovementPruningModifier
+        # movement pruning does not support global_sparsity
+        del pruning_args["global_sparsity"]
```

Raise exception if `global_sparsity==True`? fail fast!

---

Done!